### DISCUSSION TITLE
Wrong desired attitude value using ext_pose function

### DISCUSSION THREAD

**Discussion by cc299792458** [bitcraze/discussions#681] [Q&A]
Hi,

I want to use the ext_pose() function to send the position and attitude data getting from motion capture.

I tried to send (x=1.0, y=2.0, z=0.2, qx=0.1076767, qy=0.1603042, qz=0.9746426, qw=0.1130491), the quaternion can transform to eular agnle in ZYX manner as roll=15 degree, pitch=10 degree, and yaw=20 degree. I continously send this command for 10 seconds.
So, when I tried to send this command to it and I think the attitude I log will be them. But actually I got the value as the figure shows, it's changing and not change to the desired values.(the scale of x-axis is mili-second) I also tried to log the position value and they works well.

I use LogConfig to log the values.
I didn't send any setpoint to crazyflie and it did not move when I do the test that I just mentioned.

So why the attitude value are wrong? Any suggestion would be appreciate!

![Screenshot from 2023-04-22 17-07-49](https://user-images.githubusercontent.com/52767591/233774844-d29f3538-c6bb-44a3-a5a5-d4b805ed9987.png)

**Comment by cc299792458** [comment]
Only the red lines are meaningful while the blue lines are useless in this test.

**Comment by krichardsson** [comment]
I'm not sure, but could it be that your external pose is "fighting" the data from the accelerometer? If this is the case, a change in the stddev of the ext pose data would change the "mix", and hence the resulting value. Another thing to try would be to physically put the CF in approximately correct pose.

I'm not sure it would explain the yaw though...
